cfpb/consumerfinance.gov
# SVG icons as a button in the Draftail rich text editor

This change adds a Draftail plugin for inline SVG icons. This is based on [Wagtail's documentation for extending Draftail](https://docs.wagtail.org/en/stable/extending/extending_draftail.html) and on other Draftail plugins that we've explored (wagtail-draftail-anchors and wagtail-footnotes).

Draftail plugins manage content in two ways, generally:

1. DraftJS contentstate
2. Database state

For SVG icons, when a user adds the icon to the rich text editor, a DraftJS Entity is created that wraps a single space (because Entities must have >0 ranges), and that Entity gets rendered in the rich text editor by our Draftail plugin decorator as an `img` tag that loads our SVG icon from its static path. If the icon doesn't exist, the image will appear broken. The space that the Entity wraps is discarded for rendering.

When the user saves the page, the DraftJS contentstate is converted to HTML(-ish) and stored in the database. At conversion time, the Entity is converted into a `<span data-icon-name="icon-name">` and the contents of the SVG is loaded via our `svg_icon()` function and stored inside that `span` tag The space that the Entity wraps is discarded for rendering..

When the page is loaded back into the rich text editor, any `span[data-icon-name]`-matching tags are converted back to DraftJS Entities that wrap a single space, which the Draftail plugin decorator will convert to an `img` tag, etc.

## How to test this PR

1. Open a page in the Wagtail admin, or create a new page
2. Add a rich text block in a StreamField that contains one
3. Use the "Icon" button in any place before, during, and after other text blocks and observe that [defined icons](https://cfpb.github.io/design-system/foundation/iconography) show, undefined icons are broken, and that the preview pane renders the icon correctly
4. Save the page with an icon, and observe that it loads back correctly in the editor and displays on the frontend.


## Screenshots
![svgtest](https://github.com/cfpb/consumerfinance.gov/assets/10562538/4a49c13e-0848-4f0d-9f5e-e288a7dd2904)

## Notes

I've tried to isolate this code as best I can, maybe with the eye to making it an independent package someday. Right now it's tied to our codebase in two ways:

- The way we load SVG icons (`svg_icon()` to load icons when converting to database state)
- The way we store SVG icons (`/static/icons` path when loading the icon in the rich text editor)

We'd need to create hooks/configuration/etc for both of these to make this independent.

## Checklist
